### DISCUSSION TITLE
Wakelock and Vibrate permission on Android?

### DISCUSSION THREAD

**Discussion by datvm** [thudugala/Plugin.LocalNotification#515] [Q&A] [answered]
Per [documentation](https://github.com/thudugala/Plugin.LocalNotification/wiki/1.-Usage-10.0.0--.Net-MAUI):

```
[assembly: UsesPermission(Manifest.Permission.WakeLock)]

//Required so that the plugin can reschedule notifications upon a reboot
[assembly: UsesPermission(Manifest.Permission.ReceiveBootCompleted)]
[assembly: UsesPermission(Manifest.Permission.Vibrate)]
[assembly: UsesPermission("android.permission.POST_NOTIFICATIONS")]

//Optional (only for Calendar and alarm clock apps)
[assembly: UsesPermission("android.permission.USE_EXACT_ALARM")]
[assembly: UsesPermission("android.permission.SCHEDULE_EXACT_ALARM")]
````

Is there any reason why Wakelock and Vibrate are needed? The only usage I can see is [here](https://github.com/thudugala/Plugin.LocalNotification/blob/a757eaccb9d196d93200348461d86084c39883b9/Source/Plugin.LocalNotification/AndroidOption/AndroidOptions.cs#L117) and I assume it's deprecated. Should the doc be updated there?

**Comment by thudugala** [comment, marked as answer]
@datvm 

// Required so that the plugin can schedule
[assembly: UsesPermission(Manifest.Permission.WakeLock)]

[assembly: UsesPermission(Manifest.Permission.Vibrate)] is needed as Minimum permissions (https://github.com/thudugala/Plugin.LocalNotification/blob/a757eaccb9d196d93200348461d86084c39883b9/Source/Plugin.LocalNotification/AndroidOption/NotificationChannelRequest.cs#L64)

**Reply by datvm** [reply]
Darn I didn't know WakeLock is needed for the scheduling, no wonder I didn't see any scheduled notification on Android (works great on iOS btw). As for Vibrate, do I need that even if I don't use or care if it vibrates or not?
